src/ui.js

```diff
@@ -40,200 +42,215 @@ the attractions of the terrain and the encounters they find there.
+        reader.readAsText(file, "UTF-8");
+    }).then(result => parseGPX(result, (err, track) => {
+        // TODO: Make an error modal
+        if (err) {
+            parseFailures.push({name: file.name, error: err});
+        } else {
+            track.filename = file.name;
+            tracks.push(track);
         }
-
-        let reader = new FileReader()
-        reader.onload = (event) => {
-            parseGPX(event.target.result, (err, track) => {
-                // TODO: Make an error modal
-                if (err) {
-                    let file = files[fileIndex - 1]
-                    parseFailures.push({name: file.name, error: err})
-                } else {
-                    tracks.push(track)
-                }
-
-                modal.progress(fileIndex)
-
-                // do the next file, but give the UI time to update.
-                window.setTimeout(loadNextFile, 1)
-            })
+    
+        modal.progress(++i);
+    })))).then(() => {
+        tracks.forEach(t => {
+            console.log('Adding track:', t.name);
+            map.addTrack(t);
+        });
+
+        if (parseFailures.length > 0) {
+            console.error('Failed files:', parseFailures);
+            alert(`Finished loading with ${parseFailures.length} failure(s)\nView console for info.`);
         }
-
-        reader.readAsBinaryString(files[fileIndex++])
-    }
-
-    loadNextFile()
+        return modal.destroy();
+    });
 }
 
 
 function handleDragOver(evt) {
-    evt.dataTransfer.dropEffect = 'copy'
-    evt.stopPropagation()
-    evt.preventDefault()
+    evt.dataTransfer.dropEffect = 'copy';
+    evt.stopPropagation();
+    evt.preventDefault();
 }
 
 
 function buildUploadModal(numFiles) {
-    function getModalContent(numLoaded) {
-        return `<h1>Reading GPX files...</h1>
-<span id="">${numLoaded} loaded of <b>${numFiles}</b>`
-    }
+    let getModalContent = numLoaded => `<h1>Reading GPX files...</h1>
+<span id="">${numLoaded} loaded of <b>${numFiles}</b>`;
 
     let modal = picoModal({
         content: getModalContent(0),
         closeButton: false,
         escCloses: false,
         overlayClose: false,
-        overlayStyles: (styles) => { styles.opacity = 0.1 }
+        overlayStyles: styles => {
+            styles.opacity = 0.1;
+        },
     })
 
-    modal.progress = (loaded) => {
-        modal.modalElem().innerHTML = getModalContent(loaded)
-    }
+    modal.progress = loaded => {
+        modal.modalElem().innerHTML = getModalContent(loaded);
+    };
 
-    return modal
+    return modal;
 }
 
 
-export function buildSettingsModal(opts, finishCallback) {
+export function buildSettingsModal(map, opts, finishCallback) {
+    let existing = opts.lineOptions.existing ? 'checked' : '';
+    let allSameColor = map.tracks.every(val => (
+        val.options.color === map.tracks[0].options.color));
+    if (map.tracks.length > 0 && !allSameColor) existing = false;
```

Comment:
Let's use some braces here to keep it consistent

Reply:
2e29fefc4511ee2da05bd0a85299e259a8f40ad9